### PR TITLE
Fix kernel_status call to check required_kernel for modules

### PR DESCRIPTION
**Pull request by utopiabound**
Find latest kernel that supports all lustre modules installed via rpm.

Fixes Issue whamcloud/integrated-manager-for-lustre#1287 

Signed-off-by: Nathaniel Clark <nclark@whamcloud.com>

### CODE REVIEW

File: chroma_agent/action_plugins/agent_updates.py [hunk 1]
```diff
@@ -166,50 +167,93 @@ def _check_HYD4050():
...
+    )
+
+
+def kernel_ok(kernel, modlist):
+    okay = True
+    kver = kernel.split("-", 1)[1]
+    return AgentShell.run(["modinfo", "-n", "-k", kver] + modlist).rc == 0
+
+
 def kernel_status():
```
Comment (johnsonw) [inline chroma_agent/action_plugins/agent_updates.py]: The complexity of this method is at 17. You could probably abstract some of the logic into other functions and call those.

File: chroma_agent/plugin_manager.py [hunk 2]
```diff
@@ -96,8 +96,8 @@ class DevicePlugin(object):
     """
 
     FAILSAFEDUPDATE = (
-        60
-    )  # We always send an update every 60 cycles (60*10)seconds - 10 minutes.
+        60  # We always send an update every 60 cycles (60*10)seconds - 10 minutes.
+    )
```
Comment (utopiabound) [inline chroma_agent/plugin_manager.py]: Black threw up on this, so I guess it gets stuck in here

File: chroma_agent/action_plugins/agent_updates.py [hunk 3]
```diff
@@ -166,50 +167,88 @@ def _check_HYD4050():
     return None
 
 
+def kver_gt(kver1, kver2, arch):
+    """
+    True if kern1 is greater than kern2
```
Comment (jgrund) [inline chroma_agent/action_plugins/agent_updates.py]:
```suggestion
    True if kver1 is greater than kver2
```

File: chroma_agent/action_plugins/agent_updates.py [hunk 4]
```diff
@@ -166,50 +167,88 @@ def _check_HYD4050():
...
+    kern is of the form: "kernel-3.10.0-1062.el7.x86_64" (`rpm -q kernel`)
```
Comment (jgrund) [inline chroma_agent/action_plugins/agent_updates.py]:
```suggestion
    kver is of the form: "kernel-3.10.0-1062.el7.x86_64" (`rpm -q kernel`)
```

**Pull request merged**